File: server/nodejs/src/infrastructure/mappers/payment-details/fraud-results-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { FraudResults } from '../../../business/domain/payments/payment-details/fraud-results';

type FraudResultsSdk = Domain.FraudResults;

export const FraudResultsMapper = {
    fromSdkResponse: (response?: FraudResultsSdk | null): FraudResults | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            fraudServiceResult: response?.fraudServiceResult,
        };
    },
};

File: server/python/app/infrastructure/mappers/payment_details/fraud_results_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.fraud_result import FraudResults
from onlinepayments.sdk.domain.fraud_results import FraudResults as FraudResultsSdk

class FraudResultsMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[FraudResultsSdk]) -> Optional[FraudResults]:
        if response is None:
            return None

        dto = FraudResults()
        dto.fraud_service_result = response.fraud_service_result

        return dto